salt/tests/unit/modules/test_metalk8s_kubernetes_cronjob.py

```diff
@@ -0,0 +1,276 @@
+                    metalk8s_kubernetes_cronjob.get_cronjob(name, namespace), result
+                )
+
+    @utils.parameterized_from_cases(YAML_TESTS_CASES["_set_cronjob_suspend"])
+    def test__set_cronjob_suspend(
+        self,
+        result,
+        name,
+        namespace,
+        suspend,
+        spec_suspend,
+    ):
+        mock_get_object = MagicMock(
+            return_value={
+                "name": name,
+                "namespace": namespace,
+                "spec": {"suspend": spec_suspend},
+                "metadata": {"uid": "1234"},
+            }
+        )
+        mock_update_object = MagicMock(return_value=None)
+
+        with patch.dict(
+            metalk8s_kubernetes_cronjob.__salt__,
+            {
+                "metalk8s_kubernetes.get_object": mock_get_object,
+                "metalk8s_kubernetes.update_object": mock_update_object,
+            },
+        ):
+
+            self.assertEqual(
+                metalk8s_kubernetes_cronjob._set_cronjob_suspend(
+                    name, namespace, suspend
+                ),
+                result,
+            )
+
+    @utils.parameterized_from_cases(YAML_TESTS_CASES["suspend_cronjob"])
+    def test_suspend_cronjob(
+        self,
+        result,
+        name="my_cronjob",
+        namespace="default",
+    ):
+        mock_get_object = MagicMock(
+            return_value={
+                "name": name,
+                "namespace": namespace,
+                "spec": {"suspend": True},
+                "metadata": {"uid": "1234"},
+            }
+        )
+        mock_update_object = MagicMock(return_value=None)
+
+        with patch.dict(
+            metalk8s_kubernetes_cronjob.__salt__,
+            {
+                "metalk8s_kubernetes.get_object": mock_get_object,
+                "metalk8s_kubernetes.update_object": mock_update_object,
+            },
+        ):
+
+            self.assertEqual(
+                metalk8s_kubernetes_cronjob.suspend_cronjob(name, namespace), result
+            )
+
+    @utils.parameterized_from_cases(YAML_TESTS_CASES["activate_cronjob"])
+    def test_activate_cronjob(
+        self,
+        result,
+        name="my_cronjob",
+        namespace="default",
+    ):
+        mock_get_object = MagicMock(
+            return_value={
+                "name": name,
+                "namespace": namespace,
+                "spec": {"suspend": False},
+                "metadata": {"uid": "1234"},
+            }
+        )
+        mock_update_object = MagicMock(return_value=None)
+
+        with patch.dict(
+            metalk8s_kubernetes_cronjob.__salt__,
+            {
+                "metalk8s_kubernetes.get_object": mock_get_object,
+                "metalk8s_kubernetes.update_object": mock_update_object,
+            },
+        ):
+
+            self.assertEqual(
+                metalk8s_kubernetes_cronjob.activate_cronjob(name, namespace),
+                result,
+            )
```

Comment:
It's duplicated tests here, to me either you test only `activate_cronjob` and `suspend_cronjob` and you mock `get_object` and `update_object` OR you test `_set_cronjob_suspend` and for `activate_cronjob` and `suspend_cronjob` you mock this `_set_cronjob_suspend`

As you want, no strong opinion here both works well